src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,168 @@
+import at.hannibal2.skyhanni.utils.inPartialSeconds
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FlowstateHelper {
+    private val config get() = SkyHanniMod.feature.mining
+    private var streakEndTimer = SimpleTimeMark.farPast()
+    private var blockBreakStreak = 0
+
+    private var display: List<Renderable>? = null
+    private var displayDirty = false
+
+    private var flowstateCache: Int? = null
+
+    private val pickobulusPattern by RepoPattern.pattern(
+        "mining.pickobulus.blockdestroy",
+        "§7Your §r§aPickobulus §r§7destroyed §r§e(?<amount>\\d+) §r§7blocks!"
+    )
+
+    enum class FlowstateElements(val label: String, var renderable: Renderable) {
+        TITLE("§d§lFlowstate Helper", Renderable.string("§d§lFlowstate Helper")),
+        TIMER("Time Remaining: §b6.71", Renderable.string("")),
+        STREAK("Streak: §7234", Renderable.string("")),
+        SPEED("§6+600⸕", Renderable.string("")),
+        ;
+
+        override fun toString() = label
+
+        companion object {
+            @JvmField
+            val defaultOption = listOf(
+                TITLE, TIMER, STREAK, SPEED
+            )
+        }
+    }
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onBlockMined(event: OreMinedEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (flowstateCache == null) return
+
+        streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+        blockBreakStreak += event.extraBlocks.values.sum()
+        createDisplay()
+        displayDirty = true
+        attemptClearDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) { //TODO: Remove once #2540 is merged
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (flowstateCache == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+            displayDirty = true
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+
+        attemptClearDisplay()
+    }
+
+    private fun attemptClearDisplay() {
+        if (streakEndTimer.isInFuture()) return
+        blockBreakStreak = 0
+        displayDirty = true
+        createDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (!config.flowstateHelper) return
+        if (flowstateCache == null) return
+
+        if (display == null || streakEndTimer.isInFuture()) createDisplay()
```

Comment:
`createDisplay()` gets called every frame, even when there is nothing to show (no flowstate active). cache it so that it updates only 10 times a second at most, and ideally dont update at all when not in use (feature enabled but no enchantment on a tool)